src/Plugin/Knn/Handler.php

```diff
@@ -178,6 +156,33 @@ private static function knnSqlQuery(Client $manticoreClient, Payload $payload, s
 			}
 		}
 
+
 		return $result;
 	}
+
+	/**
+	 * @param Payload $payload
+	 * @param string $queryVector
+	 * @return void
+	 */
+	private static function substituteParsedQuery(Payload $payload, string $queryVector): void {
+
+		$parsedQuery = $payload::$sqlQueryParser::getParsedPayload();
+		if (!is_array($parsedQuery)) {
```

Comment:
In what case it's not array? Failed to parse?

Reply:
Yes, I changed this condition to comparison with null